src/Altinn.App.Api/Controllers/DataController.cs

```diff
@@ -452,6 +464,53 @@ public async Task<ActionResult<DataPatchResponse>> PatchFormData(
         [FromBody] DataPatchRequest dataPatchRequest,
         [FromQuery] string? language = null
     )
+    {
+        var request = new DataPatchRequestMultiple()
+        {
+            Patches = new() { [dataGuid] = dataPatchRequest.Patch },
+            IgnoredValidators = dataPatchRequest.IgnoredValidators
+        };
+        var response = await PatchFormDataMultiple(org, app, instanceOwnerPartyId, instanceGuid, request, language);
+
+        if (response.Result is OkObjectResult { Value: DataPatchResponseMultiple newResponse })
+        {
+            // Map the new response to the old response
+            return Ok(
+                new DataPatchResponse()
+                {
+                    ValidationIssues = newResponse.ValidationIssues,
+                    NewDataModel = newResponse.NewDataModels[dataGuid],
+                }
+            );
+        }
+
+        // Return the error object unchanged
+        return response.Result ?? throw new InvalidOperationException("Response is null");
+    }
+
+    /// <summary>
+    /// Updates an existing form data element with patches to multiple data elements.
+    /// </summary>
+    /// <param name="org">unique identifier of the organisation responsible for the app</param>
+    /// <param name="app">application identifier which is unique within an organisation</param>
+    /// <param name="instanceOwnerPartyId">unique id of the party that is the owner of the instance</param>
+    /// <param name="instanceGuid">unique id to identify the instance</param>
+    /// <param name="dataPatchRequest">Container object for the <see cref="JsonPatch" /> and list of ignored validators</param>
+    /// <param name="language">The language selected by the user.</param>
+    /// <returns>A response object with the new full model and validation issues from all the groups that run</returns>
+    [Authorize(Policy = AuthzConstants.POLICY_INSTANCE_WRITE)]
+    [HttpPatch("")]
+    [ProducesResponseType(typeof(DataPatchResponseMultiple), 200)]
+    [ProducesResponseType(typeof(ProblemDetails), 409)]
+    [ProducesResponseType(typeof(ProblemDetails), 422)]
```

Comment:
Codepath below produces 404 and 400 as well I think

Reply:
Fixed